src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,42 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final List<E> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final Lock readLock = lock.readLock();
+    private final Lock writeLock = lock.writeLock();
 
     public void add(E element) {
-        // write your code here
+        writeLock.lock();
+        try {
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        readLock.lock();
+        E item;
+        try {
+            item = list.get(index);
+        } finally {
+            readLock.unlock();
+        }
+        return item;
```

Comment:
```suggestion
        readLock.lock();
        try {
            return list.get(index);
        } finally {
            readLock.unlock();
        }
```

Reply:
this will be more readable :)

Reply:
Agreed, made in the right way in the next tasks, but forgot to change in this one